node/flatpak-node-generator.py

```diff
@@ -619,11 +652,51 @@ def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
         raise NotImplementedError()
 
 
+class RCFileProvider:
+    def get_node_headers(self, rcfile: Path) -> NodeHeaders:
+        raise NotImplementedError()
+
+
 class ModuleProvider(contextlib.AbstractContextManager):
     async def generate_package(self, package: Package) -> None:
         raise NotImplementedError()
 
 
+class NodeHeadersProvider(contextlib.AbstractContextManager):
```

Comment:
The AbstractContextManager in other classes was so that stuff could be done in `__exit__`, that's not used here so you can just drop the context manager / `__exit__` / the `with` statement later on